### PR TITLE
fix doc environment references init scripts

### PR DESCRIPTION
**Pull request by reinsch82**
- add a link to the introduction of the GRADLE_HOME
- use $ to indicate it is an environment variable

Fixes: #22750

### CODE REVIEW

**Comment by reinsch82** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc [hunk 1]
```diff
@@ -40,9 +40,9 @@ One main limitation of init scripts is that they cannot access classes in the `b
 There are several ways to use an init script:
 
 * Specify a file on the command line. The command line option is `-I` or `--init-script` followed by the path to the script. The command line option can appear more than once, each time adding another init script. The build will fail if any of the files specified on the command line does not exist.
-* Put a file called `init.gradle` (or `init.gradle.kts` for Kotlin) in the `__USER_HOME__/.gradle/` directory.
```
Comment (eskatos) [inline subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc:43]: It think we need to define what `USER_HOME` is. There's nothing as a `USER_HOME` environment variable.
On unices it is `$HOME`, on Windows I don't know.

Same for `GRADLE_HOME`.

Defining these upfront would be a good place to add a link to the installation documentation.

File: subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc [hunk 2]
```diff
@@ -40,9 +40,9 @@ One main limitation of init scripts is that they cannot access classes in the `b
...
-* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `__USER_HOME__/.gradle/init.d/` directory.
-* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `__GRADLE_HOME__/init.d/` directory, in the Gradle distribution. This allows you to package up a custom Gradle distribution containing some custom build logic and plugins. You can combine this with the <<gradle_wrapper.adoc#gradle_wrapper,Gradle wrapper>> as a way to make custom logic available to all builds in your enterprise.
+* Put a file called `init.gradle` (or `init.gradle.kts` for Kotlin) in the `__$USER_HOME__/.gradle/` directory.
```
Comment (aSemy) [inline subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc:43]: Should `GRADLE_USER_HOME` be in this list? Is location searched for init scripts?

Comment (eskatos) [reply subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc:43]: No, init scripts are not searched for in `GRADLE_USER_HOME`

Comment (eskatos) [reply subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc:43]: Yes.
The default for `GRADLE_USER_HOME` is `$HOME/.gradle/`

File: subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc [hunk 3]
```diff
@@ -40,9 +40,9 @@ One main limitation of init scripts is that they cannot access classes in the `b
...
-* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `__GRADLE_HOME__/init.d/` directory, in the Gradle distribution. This allows you to package up a custom Gradle distribution containing some custom build logic and plugins. You can combine this with the <<gradle_wrapper.adoc#gradle_wrapper,Gradle wrapper>> as a way to make custom logic available to all builds in your enterprise.
+* Put a file called `init.gradle` (or `init.gradle.kts` for Kotlin) in the `__$<<directory_layout.adoc#dir:gradle_user_home,GRADLE_USER_HOME>>__/` directory.
+* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `__$<<directory_layout.adoc#dir:gradle_user_home,GRADLE_USER_HOME>>__/init.d/` directory.
+* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `$<<installation.adoc#sec:linux_macos_users_2,__GRADLE_HOME__>>/init.d/` directory, in the Gradle distribution. This allows you to package up a custom Gradle distribution containing some custom build logic and plugins. You can combine this with the <<gradle_wrapper.adoc#gradle_wrapper,Gradle wrapper>> as a way to make custom logic available to all builds in your enterprise.
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/running-builds/init_scripts.adoc:45]: nitpick:
```suggestion
* Put a file that ends with `.gradle` (or `.init.gradle.kts` for Kotlin) in the `$<<installation.adoc#sec:linux_macos_users_2,__GRADLE_HOME__>>/init.d/` directory, in the Gradle distribution. This allows you to package a custom Gradle distribution containing custom build logic and plugins. You can combine this with the <<gradle_wrapper.adoc#gradle_wrapper,Gradle wrapper>> as a way to make custom logic available to all builds in your enterprise.
```

**Review by hythloda — APPROVED** [review]

**Comment by reinsch82** [comment]
@bot-gradle test and merge

**Pull request merged**